gorm.go

```diff
@@ -347,8 +347,10 @@ func (db *DB) Callback() *callbacks {
 
 // AddError add error to db
 func (db *DB) AddError(err error) error {
-	if errTranslator, ok := db.Dialector.(ErrorTranslator); ok {
-		err = errTranslator.Translate(err)
+	if err != nil {
+		if errTranslator, ok := db.Dialector.(ErrorTranslator); ok {
+			err = errTranslator.Translate(err)
+		}
```

Comment:
Move the below code to the if-check?

Reply:
Sorry, I didn't get it, do you mean adding it to the if & if else checks?

Reply:
change it to something like:

```go
func (db *DB) AddError(err error) error {
	if err != nil {
		if errTranslator, ok := db.Dialector.(ErrorTranslator); ok {
			err = errTranslator.Translate(err)
		}

		if db.Error == nil {
			db.Error = err
		} else {
			db.Error = fmt.Errorf("%v; %w", db.Error, err)
		}
	}
	return db.Error
}
```